### ISSUE TITLE
Split brain with 3 node cluster

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1018]
Originally reported of Google Code with ID 962.

```
gnt-cluster (ganeti v2.11.5) 2.11.5

Software version: 2.11.5
Internode protocol: 2110000
Configuration format: 2110000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.11.5

hspace (ganeti) version v2.11.5
compiled with ghc 7.4
running on linux x86_64

Distribution: Debian 7.6

<b>What steps will reproduce the problem?</b>

3 node cluster: A, B and C - A is master

There is one DRBD instance running primary on A, secondary on C.

I carry out a gnt-cluster master-failover from B - no problem,
everyone agrees B is the new master, and attempts to execute gnt-*
commands on A and C report that I'm not on the master node.

The issue I'm seeing:

- kill node B (halt)
- don't gnt-node --offline the node, and don't remove it from
  the cluster
- back on node A, run gnt-cluster master-failover
- after a little while, the commands returns, and node A is indeed
  convinced it's the master. node C confirms this (gnt-cluster getmaster
  returns node A in both cases)

- reboot node B
- when node B comes up, the ganeti master daemon doesn't come up:

ganeti-masterd...CRITICAL:root:It seems we are not the master (top-voted node is nodeA with 2 out of 2 votes)

So far so good, but I can still run gnt-* commands on this node (node B).

More worrying, if I issue a ganeti restart on node A:
 
ganeti-masterd...CRITICAL:root:It seems we are not the master (top-voted node is nodeB with 1 out of 2 votes)

- node C, meanwhile, is still convinced node A is master.

According to the manpage for gnt-cluster:

| If the original master then comes up, it won't be able to start its master daemon  because  it
| won't  have enough votes, but so won't the new master, if the master daemon ever needs a restart.

But now:

- node A claims it's the master (as reported by gnt-cluster getmaster)
  ... but ganeti-masterd won't restart cause it thinks B is master

- node B claims it's the master (as reported by gnt-cluster getmaster)
  ... but ganeti-masterd won't restart cause it thinks A is master

- both hosts allow me to run gnt-* commands

- node C thinks A is the master

That shouldn't happen I believe ?
```
Originally added on 2014-09-26 16:43:46 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-01-07 18:20:07 +0000 UTC.

Added Labels: Type-Defect Priority-High
Changed State: Accepted
Added to Milestone: Release2.12

**Comment by Ganeti-Issues-Migrator** [comment]
```
That bug report was for a 2.11 version; let's try to fix the issue
on that branch.
```
Originally added on 2015-01-08 14:18:35 +0000 UTC.

Added to Milestone: Release2.11
Removed from Milestone: Release2.12

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 5e641d0a518636007c2b9d6d96a18d383b2958ab
Author: Klaus Aehlig <aehlig@google.com>
Date:   Thu Jan 8 16:16:53 2015 +0100

    Also count your own vote
    
    Remove the optimization not to ask yourself as you know
    the vote already. While the effort for an additional request
    is not relevant on startup, not counting your own vote can
    lead to an blocked situation in a three-node cluster.
    Fixes issue 962.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2015-01-08 15:37:15 +0000 UTC.

Mentioned Issue with GoogleCode ID 962 has been migrated to GitHub with ID #1018.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Also verified that the voting works in 2.12 as well.
In 2.12, however, gnt-* commands do not fail immediately
if not on master (this is due to retrying operations
to to connect luxid).
```
Originally added on 2015-01-08 17:05:18 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:05 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:42 +0000 UTC.

Changed State: Released
